src/sentry/tagstore/snuba/backend.py

```diff
@@ -388,30 +390,46 @@ def get_groups_user_counts(self, project_id, group_ids, environment_id):
         result = snuba.query(start, end, ['issue'], None, filters, aggregations)
         return defaultdict(int, result.items())
 
-    # Search
-    def get_group_ids_for_search_filter(self, project_id, environment_id, tags):
-        from sentry.search.base import ANY, EMPTY
-        start, end = self.get_time_range()
-        # Any EMPTY value means there can be no results for this query so
-        # return an empty list immediately.
-        if any(val == EMPTY for _, val in six.iteritems(tags)):
-            return []
+    def get_group_ids_for_search_filter(
+            self, project_id, environment_id, tags, start, end, candidates=None, limit=1000):
+
+        from sentry.search.base import ANY
 
         filters = {
-            'environment': [environment_id],
             'project_id': [project_id],
         }
 
+        if environment_id is not None:
+            filters['environment'] = [environment_id]
+
+        # only passed in when using the DjangoSearchBackend
+        if candidates:
+            hashes = list(GroupHash.objects.filter(
+                group_id__in=candidates
+            ).values_list(
+                'hash', flat=True
+            ).distinct())
+
+            if not hashes:
+                return []
+
+            filters['primary_hash'] = hashes
+
         conditions = []
         for tag, val in six.iteritems(tags):
             col = 'tags[{}]'.format(tag)
             if val == ANY:
-                conditions.append((col, 'IS NOT NULL', None))
+                conditions.append((col, '!=', ''))
```

Comment:
Buhhh, crap. I changed it because of @alex-hofsteede's comment here: https://github.com/getsentry/sentry/pull/7925#pullrequestreview-111822758

The "does this tag exist" function we're using returns a `''` for no. Is it even possible to search for `''`? I'll have to try.